lib/views/home_content/map/post_map.dart

```diff
@@ -33,6 +39,27 @@ class PostMap extends ConsumerWidget {
       loading: () => (),
     );
 
+    //list of Marker displayed on the map
+    Set<Marker> markers = {};
+
+    mapPins.when(
+      data: (data) {
+        markers = data
+            .map(
+              (pin) => Marker(
+                markerId: pin.id,
+                position: pin.position,
+                onTap: pin.callbackFunction,
+              ),
+            )
+            .toSet();
+      },
+      error: (error, _) {
+        throw Exception("Map pins error: $error");
+      },
+      loading: () => (),
+    );
+
```

Comment:
I was wondering if we could use a `CircularValue` here to manage a possible error.
I am not entirely sure however that this is the behavior we want and we may want to ask other members of the team, what do you think ?

After a little testing I think that with a `CircularValue` the code would look like:

```dart
    return CircularValue(
      value: mapPinsAsync,
      builder: (context, mapPins) {
        //list of Marker displayed on the map
        final markers = mapPins
            .map(
              (pin) => Marker(
                markerId: pin.id,
                position: pin.position,
                onTap: pin.callbackFunction,
              ),
            )
            .toSet();

        return Expanded(
          child: GoogleMap(
            key: postMapKey,
            mapType: MapType.normal,
            myLocationButtonEnabled: true,
            myLocationEnabled: true,
            zoomGesturesEnabled: true,
            zoomControlsEnabled: true,
            scrollGesturesEnabled: true,
            rotateGesturesEnabled: false,
            tiltGesturesEnabled: false,
            initialCameraPosition: CameraPosition(
              target: mapInfo.initialLocation,
              zoom: initialZoomLevel,
            ),
            circles: mapNotifier.circles,
            markers: markers,
            onMapCreated: mapNotifier.onMapCreated,
          ),
        );
      },
    );
```

Reply:
Fixed with [65fcf16](https://github.com/ProximaEPFL/proxima/pull/231/commits/65fcf1659fde502d85d00cd3998acf4720de4054)